core/math/src/main/java/org/arakhne/afc/math/geometry/d2/afp/BasicPathShadow2afp.java

```diff
@@ -304,6 +255,79 @@ private void discretizePathIterator(
+
+    @SuppressWarnings({"checkstyle:parameternumber", "checkstyle:linelength"})
+    private void setQuadToFromDiscretizePathIterator(PathElement2afp element, PathIterator2afp<?> pi, double x1, double y1, double x2, double y2, double curx, double cury, double endx, double endy) {
+        final Path2afp<?, ?, ?, ?, ?, ?> localPath;
+        localPath = pi.getGeomFactory().newPath(pi.getWindingRule());
+        localPath.moveTo(curx, cury);
+        localPath.quadTo(
+                element.getCtrlX1(), element.getCtrlY1(),
+                endx, endy);
+        discretizePathIterator(
+                localPath.getPathIterator(MathConstants.SPLINE_APPROXIMATION_RATIO),
+                x1, y1, x2, y2);
+        if (this.crossings == MathConstants.SHAPE_INTERSECTS) {
+            return;
+        }
+    }
+
+    @SuppressWarnings({"checkstyle:parameternumber", "checkstyle:linelength"})
+    private void setCurveToFromDiscretizePathIterator(PathElement2afp element, PathIterator2afp<?> pi, double x1, double y1, double x2, double y2, double curx, double cury, double endx, double endy) {
+        final Path2afp<?, ?, ?, ?, ?, ?> localPath;
+        localPath = pi.getGeomFactory().newPath(pi.getWindingRule());
+        localPath.moveTo(curx, cury);
+        localPath.curveTo(
+                element.getCtrlX1(), element.getCtrlY1(),  element.getCtrlX2(), element.getCtrlY2(),
+                endx, endy);
+        discretizePathIterator(
+                localPath.getPathIterator(MathConstants.SPLINE_APPROXIMATION_RATIO),
+                x1, y1, x2, y2);
+        if (this.crossings == MathConstants.SHAPE_INTERSECTS) {
+            return;
+        }
+    }
+
+    @SuppressWarnings({"checkstyle:parameternumber", "checkstyle:linelength"})
+    private void setArcToFromDiscretizePathIterator(PathElement2afp element, PathIterator2afp<?> pi, double x1, double y1, double x2, double y2, double curx, double cury, double endx, double endy) {
+        final Path2afp<?, ?, ?, ?, ?, ?> localPath;
+        localPath = pi.getGeomFactory().newPath(pi.getWindingRule());
+        localPath.moveTo(curx, cury);
+        localPath.arcTo(
+                endx, endy,
+                element.getRadiusX(), element.getRadiusY(),
+                element.getRotationX(), element.getLargeArcFlag(),
+                element.getSweepFlag());
+        discretizePathIterator(
+                localPath.getPathIterator(MathConstants.SPLINE_APPROXIMATION_RATIO),
+                x1, y1, x2, y2);
+        if (this.crossings == MathConstants.SHAPE_INTERSECTS) {
+            return;
+        }
+    }
+
+    private void setCloseFromDiscretizePathIterator(double x1, double y1, double x2, double y2, double curx, double cury, double movx, double movy) {
+        if (cury != movy || curx != movx) {
+            crossSegmentTwoShadowLines(
+                    curx, cury,
+                    movx, movy,
+                    x1, y1, x2, y2);
+        }
+        if (this.crossings != 0) {
```

Comment:
This conditional statement has not the same effect as in the previous version of the code. Indeed, this new function returns whatever the value of the condition.